lib/sectorbuilder/scrub.go

```diff
@@ -0,0 +1,84 @@
+package sectorbuilder
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	sectorbuilder "github.com/filecoin-project/filecoin-ffi"
+	"golang.org/x/xerrors"
+)
+
+type Fault struct {
+	SectorID uint64
+
+	Err error
+}
+
+func (sb *SectorBuilder) Scrub(sectorSet sectorbuilder.SortedPublicSectorInfo) []*Fault {
+	var faults []*Fault
+
+	for _, sector := range sectorSet.Values() {
+		err := sb.checkSector(sector.SectorID)
+		if err != nil {
+			faults = append(faults, &Fault{SectorID: sector.SectorID, Err: err})
+		}
+	}
+
+	return faults
+}
+
+func (sb *SectorBuilder) checkSector(sectorID uint64) error {
+	cache, err := sb.sectorCacheDir(sectorID)
+	if err != nil {
+		return xerrors.Errorf("getting sector cache dir: %w", err)
+	}
+
+	if err := assertFile(filepath.Join(cache, "p_aux"), 96, 96); err != nil {
+		return err
+	}
+	if err := assertFile(filepath.Join(cache, "sc-01-data-tree-r-last.dat"), (2*sb.ssize)-32, (2*sb.ssize)-32); err != nil {
```

Comment:
is this `01` there for all sectors?

Reply:
If you have a 32G miner, would be great to verify

Reply:
I'll try that soon, just got new ram

Reply:
Just confirmed